### ISSUE TITLE
How to manually load the hmms?

### ISSUE THREAD

**Issue by Rooksie** [virtool/virtool-hmm#2]
I am unable to get the HMMs to download from within the GUI of virtools, and I am hoping you can point me at what code to change to get my virtool instance to recognize the manually downloaded HMMs for use.  I wasn't able to get the Reference file automatically either, but I was able to import the downloaded copy from the other github, install and index it without issue.

I click install, no progress happens on the bar, I tried deleting all HMM data (through settings), and restarted the install mutple times. The same things happens it just sits there. I have left it overnight and over the weekend before today and have not been successful.

virtool.log file
2022-12-09 16:18:28,041 app         INFO     Virtool v4.2.1
2022-12-09 16:18:28,043 app         INFO     Checking application data
2022-12-09 16:18:28,090 utils       INFO     Found MongoDB 4.4.16.
2022-12-09 16:18:28,104 app         INFO     Configuring Sentry
2022-12-09 16:18:28,127 app         INFO     Starting database checks. Do not interrupt. This may take several minutes.
2022-12-09 16:18:28,128 app         INFO     Checking database...
2022-12-09 16:18:28,128 migrate     INFO      • analyses
2022-12-09 16:18:28,150 migrate     INFO      • files
2022-12-09 16:18:28,159 migrate     INFO      • groups
2022-12-09 16:18:28,163 migrate     INFO      • jobs
2022-12-09 16:18:28,165 migrate     INFO      • sessions
2022-12-09 16:18:28,168 migrate     INFO      • status
2022-12-09 16:18:28,190 migrate     INFO      • subtraction
2022-12-09 16:18:28,200 app         INFO     Creating database indexes...
2022-12-09 16:18:28,290 app         INFO     Listening at http://localhost:9950
2022-12-09 16:18:31,991 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:31 +0000] "GET /api/hmms?find=&page=1 HTTP/1.1" 200 920 "http://localhost:9950/samples" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:31,992 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:31 +0000] "GET /api/indexes?ready=true HTTP/1.1" 200 807 "http://localhost:9950/samples" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:32,005 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:31 +0000] "GET /api/samples?find=&pathoscope=true&pathoscope=false&pathoscope=ip&nuvs=true&nuvs=false&nuvs=ip&page=1 HTTP/1.1" 200 794 "http://localhost:9950/samples" "Mozilla/5.0 (X11; Linux x86_
64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:32,110 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:32 +0000] "GET /api/subtractions?short=true HTTP/1.1" 200 698 "http://localhost:9950/samples" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:35,390 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:35 +0000] "GET /api/refs?find=&page=1 HTTP/1.1" 200 1332 "http://localhost:9950/refs" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:36,460 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:36 +0000] "GET /api/hmms?find=&page=1 HTTP/1.1" 200 920 "http://localhost:9950/hmm" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:42,855 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:42 +0000] "DELETE /api/hmms HTTP/1.1" 204 678 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:42,902 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:42 +0000] "GET /api/hmms?find=&page=1 HTTP/1.1" 200 906 "http://localhost:9950/hmm" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:18:44,054 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:44 +0000] "POST /api/hmms/status/updates HTTP/1.1" 400 771 "http://localhost:9950/hmm" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,294 ws          INFO     Connection closed
2022-12-09 16:19:05,299 web_log     INFO     127.0.0.1 [09/Dec/2022:21:18:28 +0000] "GET /ws HTTP/1.1" 101 0 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,343 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /hmm/settings HTTP/1.1" 200 1821 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,397 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /static/app.2e929560.js HTTP/1.1" 304 680 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,402 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /static/1.app.2e929560.js HTTP/1.1" 304 680 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,539 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/settings HTTP/1.1" 200 1282 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,542 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/account HTTP/1.1" 200 1244 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,600 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/processes HTTP/1.1" 200 34743 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,713 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/account HTTP/1.1" 200 1244 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,719 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/settings HTTP/1.1" 200 1282 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,735 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/processes HTTP/1.1" 200 34743 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:05,797 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:05 +0000] "GET /api/software HTTP/1.1" 200 887 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:10,440 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:10 +0000] "DELETE /api/hmms HTTP/1.1" 204 678 "http://localhost:9950/hmm/settings" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:10,505 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:10 +0000] "GET /api/hmms?find=&page=1 HTTP/1.1" 200 906 "http://localhost:9950/hmm" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"
2022-12-09 16:19:12,464 web_log     INFO     127.0.0.1 [09/Dec/2022:21:19:12 +0000] "POST /api/hmms/status/updates HTTP/1.1" 400 771 "http://localhost:9950/hmm" "Mozilla/5.0 (X11; Linux x86_64; rv:91.0) Gecko/20100101 Firefox/91.0"

**Comment by igboyes** [comment]
It is not possible to manually load the HMMs.

This problem is related to an incompatibility between Virtool and the host system. It has proven to be impossible to solve with our current installation method.

We do have a Docker image available that has solved the problem for other users. I can provide guidance. Just let me know:
* Are familiar with Docker?
* Have you have already loaded data into your Virtool instance?

**Comment by Rooksie** [comment]
Hello @igboyes, Happy New Year! 

I actually tried using the docker image last year before posting this question and I was unsuccessful. I wasn't able to get the virtool instance to run. I even tried many different versions of the image. There was always some error about a conflict or not being able to use something already in use. I am paraphrasing, but it was a very frustrating message that I didn't know how to fix.

- Prior to my attempt to use the virtool docker image, I had never tried/used docker images before. 
- I have successfully loaded samples and loaded and indexed references and subtractions into my virtool instance (non-docker install). I was very proud of being able to figure that out.

If additional assistance is available (I didn't see troubleshooting of the docker image before), I am willing to try the docker image again.

Thank you,
Rooksie 
![Screen Shot 2023-01-03 at 9 26 16 AM](https://user-images.githubusercontent.com/15877967/210376912-b6262264-5d7d-469a-80e6-730b08445f97.png)
![Screen Shot 2023-01-03 at 9 26 23 AM](https://user-images.githubusercontent.com/15877967/210376939-9e6d4875-aec7-411c-860a-b6c05d1db731.png)
